File: packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/types/WbMode.java
package io.flutter.plugins.camera.types;

// Mirrors wb_mode.dart
public enum WbMode {
  auto("auto"),
  incandescent("incandescent"),
  fluorescent("fluorescent"),
  warm_fluorescent("warm_fluorescent"),
  daylight("daylight"),
  cloudy_daylight("cloudy_daylight"),
  twilight("twilight"),
  shade("shade");
  
  private final String strValue;

  WbMode(String strValue) {
    this.strValue = strValue;
  }

  public static WbMode getValueForString(String modeStr) {
    for (WbMode value : values()) {
      if (value.strValue.equals(modeStr)) return value;
    }
    return null;
  }

  @Override
  public String toString() {
    return strValue;
  }
}

File: packages/camera/camera_platform_interface/lib/src/types/wb_mode.dart
enum WbMode {
  auto,
  incandescent,
  fluorescent,
  warm_fluorescent,
  daylight,
  cloudy_daylight,
  twilight,
  shade,
}

String serializeWbMode(WbMode exposureMode) {
  switch (exposureMode) {
    case WbMode.auto:
      return 'auto';
    case WbMode.incandescent:
      return 'incandescent';
    case WbMode.fluorescent:
      return 'fluorescent';
    case WbMode.warm_fluorescent:
      return 'warm_fluorescent';
    case WbMode.daylight:
      return 'daylight';
    case WbMode.cloudy_daylight:
      return 'cloudy_daylight';
    case WbMode.twilight:
      return 'twilight';
    case WbMode.shade:
      return 'shade';
    default:
      throw ArgumentError('Unknown WbMode value');
  }
}

/// Returns the exposure mode for a given String.
WbMode deserializeWbMode(String str) {
  switch (str) {
    case "auto":
      return WbMode.auto;
    case "incandescent":
      return WbMode.incandescent;
    case "fluorescent":
      return WbMode.fluorescent;
    case "warm_fluorescent":
      return WbMode.warm_fluorescent;
    case "daylight":
      return WbMode.daylight;
    case "cloudy_daylight":
      return WbMode.cloudy_daylight;
    case "twilight":
      return WbMode.twilight;
    case "shade":
      return WbMode.shade;
    default:
      throw ArgumentError('"$str" is not a valid WbMode value');
  }
}
